main.js

```diff
@@ -0,0 +1,1512 @@
+    input.value = new Date().toISOString().substr(0, 10);
+    input.classList.add("add-task-deadline");
+    input.addEventListener("change", validateInput, false);
+    contentContainer.insertBefore(input, paragraph);
+  }
+
+  const addEvent = (eventName) => {
+    messageBoxContainer.addEventListener("click", eventName, false);
+  };
+
+  const removeEvent = (eventName) => {
+    messageBoxContainer.removeEventListener("click", eventName, false);
+  };
+
+  const closeMessageBox = (evToRemove) => {
+    changeDisplayState.hide("overlay-sm", "modal-sm");
+    removeEvent(evToRemove);
+    clearElement(contentContainer);
+    renderAndReOpen();
+  };
+
+  const openMessageBox = () => {
+    changeDisplayState.show("overlay-sm", "modal-sm");
+  };
+
+  const isCancelMessageButton = function (e) {
+    return e.target.dataset.messageButton === "cancel";
+  };
+
+  const isConfirmMessageButton = function (e) {
+    return e.target.dataset.messageButton === "confirm";
+  };
+
+  function disableConfirmButton(state) {
+    const stateStr = state === true ? "disabled" : "confirm";
+    confirmButton.disabled = state;
+    confirmButton.dataset.messageButton = stateStr;
+  }
+
+  const valuesOfSendTaskTo = function () {
+    let _newContainer;
+    let _currentTaskId;
+    return {
+      get container() {
+        return _newContainer;
+      },
+      get taskId() {
+        return _currentTaskId;
+      },
+      set container(_newContainerName) {
+        _newContainer = _newContainerName;
+      },
+      set taskId(_selectedTaskId) {
+        _currentTaskId = _selectedTaskId;
+      },
+    };
+  };
+
+  const getNewStatus = function () {
+    const containerName = valuesOfSendTaskTo.container;
+    const containerStr = containerName;
+    const newStatus = containerStr.replace("-tasks", "");
+    return newStatus;
+  };
+
+  const getTaskvalues = function () {
+    const taskCardId = valuesOfSendTaskTo.taskId;
+    const containerName = valuesOfSendTaskTo.container;
+    const selectedTask = tasks.find((task) => task.id === taskCardId);
+    const currentDeadline = selectedTask.deadline;
+    return { selectedTask, containerName, currentDeadline };
+  };
+
+  const isStatusMatchesWithDateRange = function (status, dateRange) {
+    return {
+      "long-term": { true: "confirm", false: "reject" },
+      urgent: { true: "reject", false: "confirm" },
+    }[status][dateRange];
+  };
+
+  const validationContent = function (status, dateResult) {
+    return {
+      "long-term": {
+        confirm: { message: "Looks Fine !", attr: "valid" },
+        reject: {
+          message: "Deadline cannot be earlier than 7 days.",
+          attr: "error",
+        },
+      },
+      urgent: {
+        confirm: { message: "Looks Fine !", attr: "valid" },
+        reject: {
+          message: "Deadline cannot be later than 7 days.",
+          attr: "error",
+        },
+      },
+    }[status][dateResult];
+  };
```

Comment:
IMO in any case it returns the same value and `status` has no usage it can be shortened as 
```suggestion
  const validationContent = function (dateResult) {
    return {      
        confirm: { message: "Looks Fine !", attr: "valid" },
        reject: {
          message: "Deadline cannot be later than 7 days.",
          attr: "error",
      },
    }[dateResult];
  };
```

Reply:
Thank you for taking the time for this pull request. For this one; there are actually two types of future tasks on this app:

-The first one is "Urgent tasks", the tasks that need to be completed before 7 days.
-The second one is the "Long term tasks", that could be completed after 7 days from now.
 
 The app relocates those tasks by comparing their deadline values with this "threshold" value(If a user creates a task today and marks its deadline as after 7 days from now, the task will appear at "Long Term" column for today. But tomorrow the user will see it on "Urgent Tasks" column.). Because of that; if a user wants to change the status of a task, it needs to provide a new "deadline" value that matches with the "threshold" value of that new status(So he/she cannot mark "3 days left" task as a long term or "30 days left until deadline" one as an urgent task). 
For this changing status situation; there are four types of scenarios as a result of user interaction: 
A: The user could change the status of the task from "Urgent" to "Long term", 
1. And changes the deadline properly as well; then "Looks Fine!" message is going to be displayed,
2. If not then the "Deadline cannot be **earlier** than 7 days!" message is going to be shown. 
B: The user could change the status of the task from "Long term" to "Urgent"
1. If the user changes the deadline properly as well; "Looks Fine!" message is going to be displayed
2. If not then the "Deadline cannot be **later** than 7 days!" message is going to be shown. 

This function expression returns objects, that contains message value and attributes of that message, by comparing status value and dateResult value (is it confirmed or rejected based on the result of the comparison as I tried to explained above 😄 ). Without the status value it could not differentiate between A and B situation, and cannot return the proper message of that situation (in this case that is A.2.), which is quite necessary.